app/javascript/admin/controllers/article_pinned_modal_controller.js

```diff
@@ -0,0 +1,51 @@
+import ModalController from './modal_controller';
+
+export default class ArticlePinnedModalController extends ModalController {
```

Comment:
this inherits from https://github.com/forem/forem/blob/main/app/javascript/admin/controllers/modal_controller.js as it's essentially a custom modal. 

A choice I made after reading https://www.betterstimulus.com/architecture/mixins.html